File: algorithms/0074.search-a-2d-matrix/solution.go
func searchMatrix(matrix [][]int, target int) bool {
	colSize := len(matrix[0])
	for _, row := range matrix {
		if row[0] <= target && target <= row[colSize-1] {
			for _, x := range row {
				if x == target {
					return true
				}
			}

			break
		}
	}

	return false
}

File: algorithms/0074.search-a-2d-matrix/solution.rs
impl Solution {
    pub fn search_matrix(matrix: Vec<Vec<i32>>, target: i32) -> bool {
        let col_size = matrix[0].len();
        for row in matrix {
            if row[0] <= target && target <= row[col_size - 1] {
                return row.contains(&target);
            }
        }

        false
    }
}
